examples/pytorch/CaptumExample/Titanic_Captum_Interpret.py

```diff
@@ -0,0 +1,330 @@
+        param = parameter.nonzero(as_tuple=False).size(0)
+        table.add_row([name, param])
+        total_params += param
+
+    return table, total_params
+
+
+def visualize_importances(
+    feature_names,
+    importances,
+    title="Average Feature Importances",
+    plot=True,
+    axis_title="Features",
+):
+    print(title)
+    feature_imp = PrettyTable(["feature_name", "importances"])
+    feature_imp_dict = {}
+    for i in range(len(feature_names)):
+        print(feature_names[i], ": ", "%.3f" % (importances[i]))
+        feature_imp.add_row([feature_names[i], importances[i]])
+        feature_imp_dict[str(feature_names[i])] = importances[i]
+    x_pos = np.arange(len(feature_names))
+    if plot:
+        fig, ax = plt.subplots(figsize=(12, 6))
+        ax.bar(x_pos, importances, align="center")
+        ax.set(title=title, xlabel=axis_title)
+        ax.set_xticks(x_pos)
+        ax.set_xticklabels(feature_names, rotation="vertical")
+        mlflow.log_figure(fig, title + ".png")
+    return feature_imp, feature_imp_dict
+
+
+def train(USE_PRETRAINED_MODEL=False):
+    net = TitanicSimpleNNModel()
+    train_features, train_labels, test_features, test_labels, feature_names = prepare()
+    USE_PRETRAINED_MODEL = dict_args["use_pretrained_model"]
+    if USE_PRETRAINED_MODEL:
+        net.load_state_dict(torch.load("models/titanic_state_dict.pt"))
+        net.eval()
+        print("Model Loaded!")
+    else:
+        criterion = nn.CrossEntropyLoss()
+        num_epochs = dict_args["max_epochs"]
+        mlflow.log_param("epochs", num_epochs)
+        mlflow.log_param("lr", dict_args["lr"])
+
+        optimizer = torch.optim.Adam(net.parameters(), lr=dict_args["lr"])
+        input_tensor = torch.from_numpy(train_features).type(torch.FloatTensor)
+        label_tensor = torch.from_numpy(train_labels)
+        for epoch in range(num_epochs):
+            output = net(input_tensor)
+            loss = criterion(output, label_tensor)
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            if epoch % 50 == 0:
+                print(
+                    "Epoch {}/{} => Train Loss: {:.2f}".format(epoch + 1, num_epochs, loss.item())
+                )
+                mlflow.log_metric(
+                    "Epoch {} Loss".format(str(epoch + 1)), float(loss.item()), step=epoch,
+                )
+        if not os.path.isdir("models"):
+            os.makedirs("models")
+            torch.save(net.state_dict(), "models/titanic_state_dict.pt")
+    summary, _ = count_model_parameters(net)
+    mlflow.log_text(str(summary), "model_summary.txt")
+    return (
+        net,
+        train_features,
+        train_labels,
+        test_features,
+        test_labels,
+        feature_names,
+    )
+
+
+def train_step(train_features):
+    train_input_tensor = torch.from_numpy(train_features).type(torch.FloatTensor)
+    out_probs = net(train_input_tensor).detach().numpy()
```

Comment:
```suggestion
def train_step(net, train_features):
    train_input_tensor = torch.from_numpy(train_features).type(torch.FloatTensor)
    out_probs = net(train_input_tensor).detach().numpy()
```

Is `net` missing?

Reply:
ah it refers to `net` in the `__main__` scope (generated by `train`). I think this is a bit confusing, isn't it?

Reply:
@harupy , Yeha  right, I will update this changes.